.github/workflows/build.yml

```diff
@@ -0,0 +1,96 @@
+name: build
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'master'
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - 'master'
+
+env:
+  RELEASE_OUT: "./release"
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
```

Comment:
Tests are also run on specific runners with respect to the target OS and the results are merged and sent to Codecov.